### ISSUE TITLE
confirm_trade_exit() vetoes stop_loss/trailing_stop_loss exits in the first hour of a trade

### ISSUE THREAD

**Issue by arakas4488-cmd** [nateemma/strategies#59]
`Framework/BaseStrategy.py`, `confirm_trade_exit()` (around lines 1043-1071):

```python
def confirm_trade_exit(
    self,
    pair: str,
    trade: Trade,
    order_type: str,
    amount: float,
    rate: float,
    time_in_force: str,
    exit_reason: str,
    current_time: datetime,
    **kwargs,
) -> bool:

    # Reject exit if trade has been open for less than 1 hour
    # (Exception for emergency exits)
    if exit_reason not in ["force_exit", "emergency_exit"]:
        ...
        duration_hours = (t_current - t_open).total_seconds() / 3600.0
        if duration_hours < 1.0:
            return False
```

`confirm_trade_exit()` is called by freqtrade before *every* exit, including stoploss and trailing-stop exits (`exit_reason == "stop_loss"` / `"trailing_stop_loss"`), not just signal-based exits. The exemption list here only covers `force_exit` and `emergency_exit` — a regular `stop_loss` exit is not exempt.

Scenario: a trade opens, and within the first hour price moves against it far enough to trigger the stop (including the ATR-adaptive stop set via `custom_stoploss` at `after_fill`, lines ~765-811). freqtrade asks `confirm_trade_exit()` for permission to close the trade; this method returns `False` because `duration_hours < 1.0`. The exit order is rejected and the position stays open past its stop.

This matters more here because every config in the repo (`config/config*.json`) has `"stoploss_on_exchange": false` — there's no exchange-side stop as a backstop, so during that first hour there is no floor on the loss at all, neither bot-side (vetoed) nor exchange-side (disabled). It applies to leveraged configs too (`config_binanceus_leveraged.json`).

freqtrade's own docs flag this exact class of bug: `confirm_trade_exit()` can prevent stoploss exits and cause outsized losses — the documented exemption is only for exchange-side liquidations, not for a strategy's own veto logic.

I checked `Framework/test_base_strategy.py::TestConfirmTradeExit` — it covers `exit_reason="signal"` and `"force_exit"`, but not `"stop_loss"` or `"trailing_stop_loss"`, so this looks like an oversight in the exemption list rather than an intentional design choice.

(Static code review, not reproduced via execution — no local freqtrade run.)

Possible fix: add `"stop_loss"` and `"trailing_stop_loss"` to the exemption list alongside `force_exit`/`emergency_exit`, since the point of a stop is that it isn't supposed to be negotiable.

**Comment by nateemma** [comment]
I forgot reply to this. I rolled in your suggestion, but it reduced returns
quite significantly (no time to recover). So, there are some controls for
allowing some time after buying before doing the (manual) stoploss checks.
freqtrade stoploss signals are obeyed

Thanks,

Phil

**Comment by arakas4488-cmd** [comment]
Thanks for the follow-up, Phil — that makes sense.

Just to close the loop: adding `stop_loss` / `trailing_stop_loss` / `stoploss_on_exchange` / `liquidation` to the exemption list in `confirm_trade_exit()` (`fdab467c5`) is exactly what the issue was about — risk exits now bypass the `duration_hours < 1.0` veto, and the pacing only holds back discretionary `exit_signal` / `roi` exits. That resolves the reported bug.

The `stoploss_grace_hours` / `stoploss_grace_level` window (`cd280d3a7`) looks like a reasonable answer to the returns hit you saw — a wide but bounded first-hours stop beats either a hard veto or a tight stop that gets knocked out by entry noise.

One thing worth being aware of rather than a change request: since every config still runs `stoploss_on_exchange: false`, that grace-window floor is entirely bot-side, and it's wider than the old `self.stoploss = -0.05`. If the process is down during the first `stoploss_grace_hours` of a trade, there's no exchange-side backstop at the grace level. Fine as a deliberate tradeoff — just flagging it.

Happy for this to be closed — the core issue is handled.

**Comment by nateemma** [comment]
FYI, I'm currently running a study on the exit mechanism and found a few
bugs, mostly exiting too soon because of interactions between my mechanism
and the freqtrade logic. Should be pushed soon

Cheers,

Phil
